### DISCUSSION TITLE
DiskCannotBeAccessed error s3 disk

### DISCUSSION THREAD

**Discussion by MirkoFarina** [spatie/laravel-medialibrary#3250] [Q&A] [answered]
Hi everyone, 
I use spatie/laravel-medialibrary v10.0. and project whit laravel 9. 
I would to change the destination for my file in S3. I have this situation : 
this is my configuration in the config>filesystems 
's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_STORAGE_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
        ]

but in my .env file i don't have setted the 'url' and the 'endpoint' .

when i try to put the file, i have this error back : 

local.ERROR: Disk named `s3` cannot be accessed {"userId":18,"exception":"[object] (Spatie\\MediaLibrary\\MediaCollections\\Exceptions\\DiskCannotBeAccessed(code: 0): Disk named `s3` cannot be accessed at /shared/httpd/chemicals/vendor/spatie/laravel-medialibrary/src/MediaCollections/Exceptions/DiskCannotBeAccessed.php:9)
[stacktrace]
#0 /shared/httpd/chemicals/vendor/spatie/laravel-medialibrary/src/MediaCollections/FileAdder.php(452): Spatie\\MediaLibrary\\MediaCollections\\Exceptions\\DiskCannotBeAccessed::create('s3')
#1 /shared/httpd/chemicals/vendor/spatie/laravel-medialibrary/src/MediaCollections/FileAdder.php(428): Spatie\\MediaLibrary\\MediaCollections\\FileAdder->processMediaItem(Object(App\\Models\\Tenant\\SafetyDatasheet), Object(App\\Models\\Tenant\\Media), Object(Spatie\\MediaLibrary\\MediaCollections\\FileAdder))
#2 /shared/httpd/chemicals/vendor/spatie/laravel-medialibrary/src/MediaCollections/FileAdder.php(344): Spatie\\MediaLibrary\\MediaCollections\\FileAdder->attachMedia(Object(App\\Models\\Tenant\\Media))
#3 /shared/httpd/chemicals/app/Http/Controllers/SafetyDatasheetController.php(137): Spatie\\MediaLibrary\\MediaCollections\\FileAdder->toMediaCollection('safety_datashee...')

I don't have any solutions for this problem, can anyone help me??

**Comment by Tiththa** [comment]
Were you able to resolve this issue? @MirkoFarina

**Comment by MirkoFarina** [comment, marked as answer]
Yes, i resolve it. 
The problem was in the AWS_URL key in .env file, and i created it whit this path: https://s3.${AWS_DEFAULT_REGION}.amazonaws.com/${AWS_STORAGE_BUCKET}/

**Reply by Tiththa** [reply]
Thank you so much

**Comment by gbrits** [comment]
Thanks @MirkoFarina this fixed my problem too, didn't realise the URL was needed. But now the images uploaded say AccessDenied.
